packages/openapi-fetch/src/index.ts

```diff
@@ -124,11 +137,8 @@ export default function createClient<Paths extends {}>(clientOptions: ClientOpti
     if (response.ok) {
       let data: any = response.body;
       if (parseAs !== "stream") {
-        try {
-          data = await response.clone()[parseAs]();
-        } catch {
-          data = await response.clone().text();
-        }
+        const cloned = response.clone();
+        data = typeof cloned[parseAs] === "function" ? await cloned[parseAs]() : await cloned.text();
```

Comment:
Minor change: don’t `catch()`; just let the error bubble up if there’s a parsing error